aggregation/txmetrics/aggregator.go

```diff
@@ -0,0 +1,364 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package txmetrics
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/codahale/hdrhistogram"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
+
+	logs "github.com/elastic/apm-server/log"
+	"github.com/elastic/apm-server/model/metadata"
+	"github.com/elastic/apm-server/model/metricset"
+	"github.com/elastic/apm-server/model/transaction"
+	"github.com/elastic/apm-server/publish"
+	"github.com/elastic/apm-server/transform"
+)
+
+const (
+	minDuration time.Duration = 0
+	maxDuration time.Duration = time.Hour
+)
+
+// Aggregator aggregates transaction durations, periodically publishing histogram metrics.
+type Aggregator struct {
+	report              publish.Reporter
+	interval            time.Duration
+	maxBuckets          int
+	hdrHistogramSigfigs int
```

Comment:
Even though it's documented in `AggregatorConfig`? I might just add that type as a field, rather than duplicating each field. i.e.

```go
type Aggregator struct {
    config AggregatorConfig

    mu               sync.RWMutex
    active, inactive *metrics
}
```

WDYT?

Reply:
sounds good